### ISSUE TITLE
I used horovod，But There was a question about global variables.

### ISSUE THREAD

**Issue by hmilxin** [uber/horovod#135]
Hi,
     I had a question about global variables.
     How to use global variables， eg： global_step.
     
     Now , Each node had its own count （global_step）.

    Happy new year!

**Comment by alsrgv** [comment]
@hmilxin, happy new year!

You should make sure that all workers start with the same `global_step`. Then, since training is synchronous, they will be incrementing `global_step` at the same rate.

For other global variables, e.g. initial training weights, `BroadcastGlobalVariables`-Hook/Callback will broadcast them from the first worker to other workers if you add it to your training script.

E.g. for [Keras](https://github.com/uber/horovod/blob/master/examples/keras_mnist.py#L80) and [TF](https://github.com/uber/horovod/blob/master/examples/tensorflow_mnist.py#L90).

**Comment by hmilxin** [comment]
@alsrgv，
        I modified “horovod/examples/tensorflow_mnist.py” .  (two notes)
       << 
        while not mon_sess.should_stop():
            # Run a training step synchronously.
            image_, label_ = mnist.train.next_batch(100)
            mon_sess.run(train_op, feed_dict={image: image_, label: label_})
            
            **#=======begin========
            gstep = mon_sess.run(global_step)
            print ('HVD : %d-------global_step:%d'% (hvd.rank(), gstep))
            #========end=========**
       >>
         The log:
        INFO:tensorflow:loss = 2.32468, step = 1
        HVD : 1-------global_step:1
        INFO:tensorflow:loss = 2.30697, step = 1
        HVD : 0-------global_step:1
        HVD : 0-------global_step:2
        HVD : 1-------global_step:2
        HVD : 0-------global_step:3
        HVD : 1-------global_step:3
        HVD : 1-------global_step:4
        HVD : 0-------global_step:4
        ... ...

        **But I hope the log:**
         INFO:tensorflow:loss = 2.32468, step = 1
        HVD : 1-------global_step:1
        INFO:tensorflow:loss = 2.30697, step = 2
        HVD : 0-------global_step:2
        HVD : 0-------global_step:3
        HVD : 1-------global_step:4
        HVD : 0-------global_step:5
        HVD : 1-------global_step:6
        HVD : 1-------global_step:7
        HVD : 0-------global_step:8
        ... ...

        Thanks!

**Comment by alsrgv** [comment]
@hmilxin, it's actually correct to have the same global step on both workers, since they're synchronously executing code.  You just need to take into account that single global step increment is equivalent to `hvd.size()` individual increments.

For this reason, you'd typically want to do two things (examples will be updated accordingly in the next doc update round):
1. `opt = tf.train.RMSPropOptimizer(0.01 * hvd.size())` - multiply LR by `hvd.size()`
2. `tf.train.StopAtStepHook(last_step=100 // hvd.size())` - divide number of training steps by `hvd.size()`.

**Comment by hmilxin** [comment]
@alsrgv  thanks！
This may not be what I need.
I need a global variable to control the overall progress, and I'll try other ways.

thank you

**Comment by hmilxin** [comment]
@alsrgv  I had another question，
   I tested the sample “tensorflow_mnist.py” 。
  run： 
mpirun -np 2 -H localhost:2 -bind-to none -map-by slot -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH ...

 Allreduce is not used here, then the parameters are how to update it ?  In our project , one note's result  is similar to two nodes。

**Comment by alsrgv** [comment]
Closing this issue due to lack of activity.
